elide-core/src/main/java/com/yahoo/elide/optimization/UserCheck.java

```diff
@@ -3,17 +3,19 @@
  * Licensed under the Apache License, Version 2.0
  * See LICENSE file in project root for terms.
  */
-package com.yahoo.elide.security;
+package com.yahoo.elide.optimization;
 
-import com.yahoo.elide.core.PersistentResource;
+import com.yahoo.elide.security.User;
 
 /**
  * Custom security access that verifies whether a user belongs to a role.
  * Permissions are assigned as a set of checks that grant access to the permission.
  *
- * @param <T> Type of record for Check
+ * NOTE: These checks are intended to be used for specific optimizations. That is, if the check
+ * only requires user-level information and can cover an entire entity. These only apply to at
+ * the entity-level. Field-level annotating of UserCheck's is not supported.
  */
-public interface UserCheck<T> extends Check<T> {
+public interface UserCheck {
```

Comment:
Minor comment. Might be nice to change function name 'userPermission' to 'ok' to match other checks.